core/server/master/src/main/java/alluxio/master/AlluxioSimpleMasterProcess.java

```diff
@@ -94,61 +69,35 @@ public abstract class AlluxioSimpleMasterProcess extends MasterProcess {
       LOG.error("Failed to create {} master", mMasterName, e);
       throw new RuntimeException(String.format("Failed to create %s master", mMasterName), e);
     }
-    try {
-      stopServing();
-    } catch (Exception e) {
-      throw new RuntimeException(e);
-    }
-  }
-
-  @Override
-  public InetSocketAddress getWebAddress() {
-    synchronized (mWebServerLock) {
-      if (mWebServer != null) {
-        return new InetSocketAddress(mWebServer.getBindHost(), mWebServer.getLocalPort());
-      }
-      return mWebConnectAddress;
-    }
-  }
-
-  @Override
-  public InetSocketAddress getRpcAddress() {
-    return mRpcConnectAddress;
   }
 
   @Override
   public void start() throws Exception {
+    mServices.forEach(SimpleService::start);
```

Comment:
If `mServices` is a HashSet then the order here is unpredictable. I'm not sure if that makes things trickier.

Reply:
SGTM. Later if we identify some implicit orders, we can add those logic then.